lib/cloudcmd/lib/server/rest.js

```diff
@@ -249,124 +249,128 @@
+
         case 'pack':
             if (!files.from)
                 callback(body);
             else
                 pack(files.from, files.to, files.names, callback);
             break;
-        
+
         case 'extract':
             if (!files.from)
                 callback(body);
             else
                 extract(files.from, files.to, callback);
-                
+
             break;
-        
+
         default:
             callback();
             break;
         }
     }
-    
+
     function pack(from, to, names, fn) {
         from    = root(from);
         to      = root(to);
-        
+
         if (!names) {
             names = [
                 path.basename(from)
             ];
-            
+
             from = path.dirname(from);
         }
-        
+
         operation('pack', from, to, names, fn);
     }
-    
+
     function extract(from, to, fn) {
         from    = root(from);
-                
+
         if (to)
             to  = root(to);
         else
             to = from.replace(/\.tar\.gz$/, '');
-        
+
         operation('extract', from, to, fn);
     }
-    
+
     function operation(op, from, to, names, fn) {
         var packer, wasError;
-        
+
         if (!fn) {
             fn      = names;
             names   = [
                 path.basename(from)
             ];
         }
-        
+
         packer = jaguar[op](from, to, names);
-        
+
         packer.on('error', function(error) {
             wasError = true;
             fn(error);
         });
-        
+
         packer.on('progress', function(count) {
             process.stdout.write(rendy('\r{{ operation }} "{{ name }}": {{ count }}%', {
                 operation   : op,
                 name        : names[0],
                 count       : count
             }));
         });
-        
+
         packer.on('end', function() {
             var name, msg;
-            
+
             process.stdout.write('\n');
-            
+
             if (!wasError) {
                 name    = path.basename(from),
                 msg     = formatMsg(op, name);
                 fn(null, msg);
             }
         });
     }
-    
-    function copy(from, to, names, fn) {
+
+    var copy = function(from, to, names, fn) {
         var error,
             tmpl    = '\r copy {{ from }} {{ to }} {{ count }}%',
             cp      = copymitter(from, to, names);
-         
+
         cp.on('error', function(e) {
             error = e;
             cp.abort();
         });
-        
+
         cp.on('progress', function(count) {
             process.stdout.write(rendy(tmpl, {
                 to: to,
                 from: from,
                 count: count
             }));
         });
-        
+
         cp.on('end', function() {
             process.stdout.write('\n');
             fn(error);
         });
-    }
-    
+    };
+
+    // expose copy function for testing purposes
+    module.exports.copy = copy;
```

Comment:
Be careful with unnecessary whitespace changes. There are so many in this file that it is difficult to see where the actual code changes are.